test/system/src/test/java/io/pravega/test/system/StreamsAndScopesManagementTest.java

```diff
@@ -211,6 +212,50 @@ private void testCreateSealAndDeleteStreams(String scope) {
         }
     }
 
+    @Test
+    public void testStreamTags(){
+        // Perform management tests with Streams and Scopes.
+        for (int i = 0; i < TEST_ITERATIONS; i++) {
+            log.info("Stream tag test in iteration {}.", i);
+            final String scope = "testStreamsTags" + i;
+            testCreateScope(scope);
+            testCreateUpdateDeleteStreamTag(scope);
+            testDeleteScope(scope);
+        }
+    }
+
+    private void testCreateUpdateDeleteStreamTag(String scope) {
+        final ClientConfig clientConfig = Utils.buildClientConfig(controllerURI);
+        final Set<String> tagSet1 = Set.of("t1", "t2", "t3");
+        final Set<String> tagSet2 = Set.of("t4", "t5", "t6");
+        List<CompletableFuture<Void>> futures = new ArrayList<>();
+        // Create and Update Streams
+        for (int j = 1; j <= TEST_MAX_STREAMS; j++) {
+            StreamConfiguration config = StreamConfiguration.builder().scalingPolicy(ScalingPolicy.fixed(j)).build();
+            final String stream = "stream"+j;
+            streamManager.createStream(scope, stream, config);
+            streamManager.updateStream(scope, stream, config.toBuilder().tags(tagSet1).build());
+            assertEquals(tagSet1, streamManager.getStreamTags(scope, stream));
+            assertEquals(singletonList(Stream.of(scope, stream)), newArrayList(streamManager.listStreams(scope, "t1")));
```

Comment:
in the subsequent iterations, the streams returned by `streamManager.listStreams` should not be a singleton list, right ?